pkgs/top-level/all-packages.nix

```diff
@@ -17896,18 +17898,21 @@ in
 
   soapyremote = callPackage ../applications/radio/soapyremote { };
 
-  soapysdr-with-plugins = callPackage ../applications/radio/soapysdr {
-    extraPackages = [
-      limesuite
-      soapyairspy
-      soapyaudio
-      soapybladerf
-      soapyhackrf
-      soapyremote
-      soapyrtlsdr
-      soapyuhd
-    ];
-  };
+  soapysdrplay = callPackage ../applications/radio/soapysdrplay { };
+
+  soapysdr-with-plugins = makeOverridable ({ sdrplaySupport ? false }:
```

Comment:
Using `makeOverridable` is not a good choice here. It masks all other inputs for possible overrides (besides failing to evaluate).
Since you need an override anyway to activate the plugin you could use `extraPackages` directly:
```
soapysdr-with-plugins = super.soapysdr-with-plugins.override {
  extraPackages = with super; [
    soapysdrplay
    <other plugins of your choice >
  ];
};
```

Reply:
Only problem with that is you have to list out all the default packages if you still want them. I think there might be a way to make it work correctly with `makeOverridable` - I'll try to look into that later today.

Reply:
I wasn't able to get that to work. You may just have to scrap the `sdrplaySupport` flag and force the user to list everything they want in `extraPackages` like @markuskowa says. Or maybe you could have a `soapysdr-with-sdrplay` entry in all-packages.nix that included all the default packages plus soapysdrplay. There might be a more elegant solution but I'm not able to come up with anything right now.

Reply:
Thanks for giving it a try too.
I would then indeed just scrap the idea of having this `sdrSupport` flag. I updated the implementation accordingly and changed documentation on the sdrplayApi module to suggest an override of the `extraPackages` argument.

Reply:
@lasandell I agree that it is a disadvantage that you now have to list all plugins explicitly in the override.
However, on an individual setup you may only need a few of them anyway. On an embedded system you might even want to keep it small.